bin/run-utils.js

```diff
@@ -184,3 +185,215 @@ function runFirefox(options) {
+}
+
+function prefValueToString(val) {
+  switch (typeof val) {
+    case "boolean":
+      return val ? "true" : "false";
+    case "number":
+      return "" + val;
+    case "string":
+      return "\"" + val.replace(/[^\\]{0}"/g, "\\\"") + "\"";
+    default:
+      return val;
+  }
+}
+
+function getUserPrefLine(name, value) {
+  return "user_pref(\"" + name + "\", " + prefValueToString(value) + ");";
+}
+
+function writeUserPrefs(profilePath, userPrefs) {
+  if (!userPrefs) {
+    userPrefs = currentSample || [];
+  }
+
+  var prefsFile = Path.join(profilePath, "user.js");
+  return Fs.writeFile(prefsFile, userPrefs.join("\n") + "\n", "utf8");
+}
+
+function setUserPrefs(profilePath, prefs) {
+  return takePrefsSample(profilePath)
+    .then(function() {
+      var seen = {};
+      var userPrefs = currentSample.map(function(entry) {
+        if (entry in currentSampleMap) {
+          // We encountered a line containing a pref.
+          var prefValue = currentSampleMap[entry];
+          if (entry in prefs) {
+            // This line of the prefs needs to be overwritten.
+            prefValue = prefs[entry];
+            seen[entry] = 1;
+          }
+
+          return getUserPrefLine(entry, prefValue);
+        }
+
+        return entry;
+      });
+
+      Object.getOwnPropertyNames(prefs).forEach(function(entry) {
+        if (seen[entry]) {
+          // An existing pref line has been overwritten above.
+          return;
+        }
+
+        userPrefs.push(getUserPrefLine(entry, prefs[entry]));
+      });
+
+      return writeUserPrefs(profilePath, userPrefs);
+    });
+}
+
+exports.setUserPrefs = setUserPrefs;
+
+function restoreUserPrefs(profilePath) {
+  return writeUserPrefs(profilePath);
+}
+
+exports.restoreUserPrefs = restoreUserPrefs;
+
+var ALPHA = "abcdefghijklmnopqrstuvwxyz1234567890";
+
+function randomString(len) {
+  var alphaLen = ALPHA.length;
+  var ret = [];
+  for (var i = 0, l = len || 8; i < l; ++i) {
+    ret.push(ALPHA.charAt(Math.random() * alphaLen));
+  }
+  return ret.join("");
+}
+
+var DEFAULT_PREFS = {
+  "browser.EULA.override": true,
+  "browser.EULA.3.accepted": true,
+  "browser.shell.skipDefaultBrowserCheck": true,
+  "general.warnOnAboutConfig": false,
+  "security.fileuri.origin_policy": 3,
+  "security.fileuri.strict_origin_policy": false,
+  "security.warn_entering_secure": false,
+  "security.warn_entering_secure.show_once": false,
+  "security.warn_entering_weak": false,
+  "security.warn_entering_weak.show_once": false,
+  "security.warn_leaving_secure": false,
+  "security.warn_leaving_secure.show_once": false,
+  "security.warn_submit_insecure": false,
+  "security.warn_viewing_mixed": false,
+  "security.warn_viewing_mixed.show_once": false,
+  "toolkit.telemetry.reportingpolicy.firstRun": false
+};
+
+function createEmptyProfile(profile) {
+  return When.promise(function(resolve, reject) {
+    var profileDir = randomString() + "." + profile;
+    var finder = new FxProfileFinder();
```

Comment:
Could we make use of the -CreateProfile command line argument instead of reimplementing some of the profile creation logic? Something like.. `require("child_process").spawn('/Applications/Firefox.app/Contents/MacOS/firefox-bin', ["-CreateProfile", "loop-dev"])`